include/msgpack/sysdep.h

```diff
@@ -51,8 +51,8 @@
 #       endif /* WIN32_LEAN_AND_MEAN */
 #   endif
     typedef long _msgpack_atomic_counter_t;
-#   define _msgpack_sync_decr_and_fetch(ptr) InterlockedDecrement(ptr)
-#   define _msgpack_sync_incr_and_fetch(ptr) InterlockedIncrement(ptr)
+#   define _msgpack_sync_decr_and_fetch(ptr) _InterlockedDecrement(ptr)
+#   define _msgpack_sync_incr_and_fetch(ptr) _InterlockedIncrement(ptr)
```

Comment:
What is the difference between `InterlockedDecrement` and `_InterlockedDecrement` ?
It seems that both function exists.

I don't know much about Windows APIs. I'd like to know `_InterlockedDecrement` is always better choice than `InterlockedDecrement`.

Reply:
> What is the difference between InterlockedDecrement and _InterlockedDecrement ?
> It seems that both function exists.
> I don't know much about Windows APIs. I'd like to know _InterlockedDecrement is always better choice than InterlockedDecrement.

https://docs.microsoft.com/en-us/previous-versions/ms919120(v=msdn.10)
https://docs.microsoft.com/en-us/cpp/intrinsics/interlockeddecrement-intrinsic-functions?view=vs-2019

seamed InterlockedDecrement is for Windows CE

Reply:
Thank you. It seems that the code should be as follows:

```cpp

#if WINCE /* Some how get WinCE or Not */
#   define _msgpack_sync_decr_and_fetch(ptr) InterlockedDecrement(ptr)
#   define _msgpack_sync_incr_and_fetch(ptr) InterlockedIncrement(ptr)
#else  /* WINCE */
#   define _msgpack_sync_decr_and_fetch(ptr) _InterlockedDecrement(ptr)
#   define _msgpack_sync_incr_and_fetch(ptr) _InterlockedIncrement(ptr)
#endif  /* WINCE */
```

Reply:
`InterlockedDecrement` and its friends are Win32 API so it is available on any Desktop Windows and UWP as well: https://docs.microsoft.com/en-us/windows/desktop/api/winnt/nf-winnt-interlockeddecrement it IS available Windows other than WinCE, but **NOT** x64.

`_InterlockedDecrement` and its friends are compiler intrinsic, which is always available on modern Windows platforms(x86, x64 and ARM32). So I guess it is okay to just ignore WinCE here as it is almost being obsolete. 

In case we need some completeness here, check against x64 (and ARM) then use `_InterlockedDecrement` only on there.

ref: https://github.com/ivmai/libatomic_ops/blob/347d9784ec721e78fb6111cceac6847728b361d4/src/atomic_ops.h#L387

Reply:
@okuoku, thank you for the comment. @RPG3D, could you update the PR?

Reply:
@redboltz  check the again

Reply:
@RPG3D, it looks good to me. Thank you for updating!